src/Measurements.jl

```diff
@@ -61,7 +61,19 @@ function Measurement(val::V, err::E, tag::UInt64,
 end
 Measurement{T}(x::Measurement{S}) where {T,S} = convert(Measurement{T}, x)
 Measurement{T}(x::S) where {T,S} = convert(Measurement{T}, x)
-Measurement{T}(x::S) where {T,S<:Rational} = convert(Measurement{T}, x)
+
+# disambiguities
+Measurement{T}(x::S) where {T, S<:Rational} = convert(Measurement{T}, x)
+Measurement{T}(x::S) where {T, S<:Complex} = convert(Measurement{T}, x)
+Measurement{T}(x::S) where {T, S<:Base.TwicePrecision} = convert(Measurement{T}, x)
+Measurement{T}(x::S) where {P, T, S<:Rational{P}} = convert(Measurement{T}, x)
+
+
+function Measurement{T}(::S) where {T, S<:AbstractChar}
+    throw(ArgumentError("cannot convert `$S` to `Measurement{$T}`"))
+end
```

Comment:
I'm surprised this is needed at all.  Or anyway `AbstractChar` looks oddly specific, why that type and not anything else?  Perhaps `S` shouldn't have any constraint (or `S<:Any`, which is the same) and this would be the generic fallback for non `Real` types?

Reply:
If you comment out this function, then you will see the following

```julia
1 ambiguities found
Ambiguity #1
Measurements.Measurement{T}(x::S) where {T, S} in Measurements at /home/roger/code/julia/Measurements/src/Measurements.jl:63
(::Type{T})(x::AbstractChar) where T<:Union{AbstractChar, Number} in Base at char.jl:50

Possible fix, define
  Measurements.Measurement{T}(::S) where {T, S<:AbstractChar}
```

I think this is due to the fact you can convert a `Char` to a Number defined in Base. And because `Base` uses `AbstractChar` so we have to use `AbstractChar`, why `S <: Any` is the same?

Reply:
actually, should it work for converting things to `Measurement`? since the following works

```julia
Float64('a')
```

it is probably strange that meanwhile

```
Measurement{Float64}('a')
```

doesn't work. I'm not sure this is useful anyway tho, so either is fine

Reply:
Oh, yes, I'm looking forward to character arithmetic with uncertainties: `'z' ± 'a'`! :grin: